api-design-guidelines/index.md

```diff
@@ -93,7 +93,9 @@ is printed.
       great summary.
 
     * **Use a single sentence fragment** if possible, ending with a
-      period.  Do not use a complete sentence.
+      period.  Do not use a complete sentence.  A summary is like 
+      a dictionary definition; it is shown in the context of the 
+      API it describes.
```

Comment:
on Thu Feb 18 2016, Brent Royal-Gordon <notifications-AT-github.com> wrote:

>> @@ -93,7 +93,9 @@ is printed.
>>        great summary.
>>  
>>      * **Use a single sentence fragment** if possible, ending with a
>> -      period.  Do not use a complete sentence.
>> +      period.  Do not use a complete sentence.  A summary is like 
>> +      a dictionary definition; it is shown in the context of the 
>> +      API it describes.
>
> I used the phrase "is shown in the context of" because I wanted to be
> vague about how it would be displayed, but I think it's fair to say
> that it'll almost always be clear which API it's talking about. That
> clarity might be because it's being shown in a popover, or at the
> bottom of an autocomplete list, or in a document with the API's name
> as a heading, or in some other way, but the point is that you can
> always assume there is something else nearby that clarifies which API
> the summary refers to.

Isn't this obvious?  I mean, the summary is in the doc comment, which is
right there next to the declaration.  You'd need tooling to separate it
somehow.

> I'm also a little worried about making "read like a dictionary
> definition" the emphasis of the rule. Dictionary definitions have
> several properties we may not want people to emulate—they're often
> concise to the point of vagueness, sometimes use very closely related
> words ("tighten: make or become tight or tighter"), and occasionally
> devolve into a list of synonyms. (More concretely, they're also
> capitalized differently from our summaries, with a lowercase first
> letter.) We don't want people taking the wrong ideas away from the
> rule.

OK, you're right.

> What I'm trying to get across is that they are like dictionary
> definitions in the particular respect that they are sentence fragments
> referring to a subject which the context already makes clear. My
> original thought was that it should say something along the lines of
> "Imagine the summary as being proceeded by the declaration it
> describes, followed by a colon; there's no need to repeat the
> subject." I just couldn't find a way to describe that which wasn't
> clumsy until I came up with the dictionary analogy.

I'm just not sure this all pulls its weight in terms of clarification.
We show lots of examples.

-- 
-Dave